packages/next/src/builders/build/lib/create-package-json.ts

```diff
@@ -18,6 +18,13 @@ export async function createPackageJson(
     ...rootPackageJson.devDependencies,
   };
 
+  const pickedDependencies = options.dependencies.reduce((acc, value) => {
```

Comment:
We can read npm dependencies from the project graph.

```
import { createProjectGraph } from '@nrwl/workspace/src/core/project-graph';


export async function createPackageJson(
  options: NextBuildBuilderOptions,
  context: BuilderContext
) {

  // ...
  const graph = createProjectGraph();
  const projectNpmDependencies = 
  graph.dependencies[context.target.project].reduce((acc, d) => {
    if (d.type === 'npm') {
      acc[d.data.packageName] = d.data.version;
    }
    return acc;
  }, {});

  // ...
}
```

I think the dependencies option still make sense if there are additional dependencies that Nx did not pick up (because they were not imported in the source). I would rename it to `extraDependencies` since we should be able to automatically detect most dependencies.